source/package-json.d.ts

```diff
@@ -305,6 +308,16 @@ declare namespace PackageJson {
 		nohoist?: WorkspacePattern[];
 	}
 
+	/**
+	An alternative configuration for ESLint.
+	*/
+	export interface EslintConfig {
+		/**
+		ESLint configuration options.
+		*/
+		eslintConfig?: Linter.Config;
```

Comment:
There is a hack where we don't have to include the ESLint module in our dependencies:

```suggestion
    // @ts-ignore
		eslintConfig?: import('eslint').Linter.Config;
```

Then it will use `import('eslint').Linter.Config` if it correctly finds it, else it will just fallback to any.

I used that pattern a lot through [`ts-ignore-import`](https://github.com/Sydsvenskan/node-ts-ignore-import) and its the one pattern I think we maybe could use to still include this here. And if we do, then we can include it in  `PackageJson`  straight away rather than in a `PackageJsonExtras`. Thoughts @sindresorhus?

Reply:
I don't think this is a good pattern. Falling back to `any` is never something I would want. It could accidentally allow specifying anything there without a warning. Maybe if it fell back to `never`.

Reply:
That's a really interesting pattern. I couldn't figure out how to override the `any` type with `never` (or anything other than `any` for that matter).

Side note, apparently `@types/eslint` is included in `@typescript-eslint` so I had to manually delete the package to test it. But yeah, probably best to leave versioning to the consumer.